modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/SharingUrlTimestampParser.kt

```diff
@@ -0,0 +1,38 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+class SharingUrlTimestampParser {
+    companion object {
+        val intervalPattern = Regex("""(\d*\.?\d*)?,?(\d*\.?\d*)?""") // t=10,20 or t=,20 or t=10
+        val hmsPattern = Regex("""(?:(\d+)h)?(?:(0?[0-5]?\d)m)?(?:(0?[0-5]?\d(?:\.\d+)?)s)?""") // t=10h20m30s or t=10m30s or or 10m or t=10s
+        val hhmmssFractionPattern = Regex("""(\d+):(\d{2}):(\d{2})(?:\.(\d+))?""") // t=10:20:30.456
```

Comment:
All of these, not only `hmsPattern`, need to handle minutes and seconds above `59`. Can you add tests for these cases as well?

Reply:
[a18bb80](https://github.com/Automattic/pocket-casts-android/pull/2235/commits/a18bb8044bbe92f1a0354d65aa2fb29ecb46d6f9)